pulser-core/pulser/waveforms.py

```diff
@@ -865,6 +860,51 @@ def __init__(
             **interpolator_kwargs,
         }
 
+    @staticmethod
+    def _check_values_times(
+        values: Union[ArrayLike, Parametrized],
+        times: Optional[Union[ArrayLike, Parametrized]] = None,
+    ) -> None:
+        """Check whether the types of values and times are valid."""
+
+        def _err_message(argument_name: str) -> str:
+            return (
+                f"`{argument_name}` must be a parametrized object or a "
+                "sequence of elements castable to float. To make a sequence"
+                " of parametrized object, instantiate a variable with a size."
+            )
+
+        if not isinstance(values, Parametrized):
+            try:
+                _values = np.array(values, dtype=float)
+            except TypeError as e:
+                raise TypeError(_err_message("values")) from e
+        if times is not None and not isinstance(times, Parametrized):
+            try:
+                times = cast(ArrayLike, times)
+                times_ = np.array(times, dtype=float)
```

Comment:
This will fail for torch tensors with requires_grad. You should use `AbstractArray` instead (and cover all possibilities in the tests)

Reply:
Nevermind, we actually don't support tensors on `InterpolatedWaveform`